.github/workflows/ossf-scorecard.yaml

```
@@ -26,12 +26,12 @@ jobs:
 
     steps:
       - name: "Checkout code"
-        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

```suggestion
        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # tag=v4.1.1
```

Let's use this format. Probably both works, but not sure

Can you please also pin the sha in golangci-lint: checkout

---

I'll open a PR

---

https://github.com/kubernetes-sigs/controller-runtime/pull/2729